src/main/java/seedu/duke/utils/Parser.java

```diff
@@ -395,58 +396,164 @@ private void createUnmarkCommand(String arguments) {
     //@@author
 
     //@@author trishaangelica
+
     /**
      * Initialises the EditCommand.
      */
     private void createEditCommand(String arguments) {
         try {
-            final Matcher matcher = ITEM_ARGS_FORMAT.matcher(arguments.trim());
-            // Validate args string format
-            if (!matcher.matches()) {
-                LOGGER.log(Level.WARNING, "(Edit command) Rejecting user command, invalid command format entered.");
-                newCommand = new IncorrectCommand(EditCommand.MESSAGE_FAILURE_INCORRECT_FORMAT);
+            final Matcher matcher = EDIT_ITEM_ARGS_FORMAT.matcher(arguments.trim());
+
+            String[] newValues;
+            final boolean iPresent = arguments.contains("i/");
+            final boolean pPresent = arguments.contains("p/");
+            final boolean qPresent = arguments.contains("q/");
+
+            boolean delimiterPresent = iPresent | pPresent | qPresent;
+            boolean noDuplicateDelimiter = duplicateDelimiterChecker(arguments);
```

Comment:
This observation was made by @JLoh579 that the coding standard recommends not to use negation in boolean variable names. 

Personally, I was thinking it would be better with something like "hasSingleDelimiter"? Just a suggestion!

Reply:
I have changed it 👍 thank you for the suggestion!